lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -151,16 +152,47 @@ export class ResponseHandler {
     /**
      * Function which validates server authorization token response.
      * @param serverResponse
+     * @param refreshAccessToken
      */
     validateTokenResponse(
-        serverResponse: ServerAuthorizationTokenResponse
+        serverResponse: ServerAuthorizationTokenResponse,
+        refreshAccessToken?: boolean
     ): void {
         // Check for error
         if (
             serverResponse.error ||
             serverResponse.error_description ||
             serverResponse.suberror
         ) {
+
+            // check if 500 error
+            if (
+                refreshAccessToken &&
+                serverResponse.status &&
+                (serverResponse.status >= ErrorCodeBounds.stsErrorLowerLimit) &&
+                (serverResponse.status <= ErrorCodeBounds.stsErrorUpperLimit)
+            ) {
+                this.logger.warning(
+                    "executeTokenRequest:validateTokenResponse - AAD is currently unavailable and the access token is unable to be refreshed."
+                );
+                
+                // don't throw an exception, but alert the user via a log that the token was unable to be refreshed
+                return;
+            // check if 400 error
+            } else if (
+                refreshAccessToken &&
+                serverResponse.status &&
+                (serverResponse.status >= ErrorCodeBounds.clientErrorLowerLimit) &&
+                (serverResponse.status <= ErrorCodeBounds.clientErrorUpperLimit)
+            ) {
+                this.logger.warning(
```

Comment:
Same as above. If a bug is raised and you see this log message, it is not sufficient to understand what's going on.